src/fmu/settings/_resources/log_manager.py

```diff
@@ -32,8 +34,52 @@ def add_log_entry(self: Self, log_entry: LogEntryType) -> None:
             )
             log_model.add_entry(validated_entry)
             self.save(log_model)
+            self._cached_dataframe = None
         except ValidationError as e:
             raise ValueError(
-                f"Invalid log entry added to '{self.__class__.__name__}' with "
+                f"Invalid log entry added to '{self.model_class.__name__}' with "
                 f"value '{log_entry}': '{e}"
             ) from e
+
+    def filter_log(self: Self, filter: Filter) -> Log[LogEntryType]:
+        """Filters the log resource with the provided filter."""
+        if self._cached_dataframe is None:
+            if not self.exists:
+                raise FileNotFoundError(
+                    f"Resource file for '{self.__class__.__name__}' not found "
+                    f"at: '{self.path}'"
+                )
+            log_model: Log[LogEntryType] = self.load()
+            df_log = pandas.DataFrame([entry.__dict__ for entry in log_model])
```

Comment:
I can see performance reasons for using `__dict__` over `model_dump()`, but curious what prompted it?